tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -9740,99 +9756,130 @@ TEST_F(ExecutionTest, DynamicResourcesUniformIndexingTest) {
-    D3D_SHADER_MODEL sm = TestShaderModels[i];
-    LogCommentFmt(L"\r\nVerifying Dynamic Resources Uniform Indexing in shader "
-                  L"model 6.%1u",
-                  ((UINT)sm & 0x0f));
+      CComPtr<ID3D12Device> pDevice;
+      if (!CreateDevice(&pDevice, sm, false /* skipUnsupported */)) {
+        continue;
+      }
+      D3D12_FEATURE_DATA_D3D12_OPTIONS devOptions;
+      VERIFY_SUCCEEDED(
+          pDevice->CheckFeatureSupport((D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS,
+                                       &devOptions, sizeof(devOptions)));
+      if (devOptions.ResourceBindingTier < D3D12_RESOURCE_BINDING_TIER_3) {
+        WEX::Logging::Log::Comment(
+            L"Device does not support Resource Binding Tier 3");
+        WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
+        return;
+      }
 
-    CComPtr<ID3D12Device> pDevice;
-    if (!CreateDevice(&pDevice, sm, false /* skipUnsupported */)) {
-      continue;
-    }
-    D3D12_FEATURE_DATA_D3D12_OPTIONS devOptions;
-    VERIFY_SUCCEEDED(
-        pDevice->CheckFeatureSupport((D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS,
-                                     &devOptions, sizeof(devOptions)));
-    if (devOptions.ResourceBindingTier < D3D12_RESOURCE_BINDING_TIER_3) {
-      WEX::Logging::Log::Comment(
-          L"Device does not support Resource Binding Tier 3");
-      WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
-      return;
-    }
+      // Add compile options
+      char compilerOptions[256];
+      if (non_uniform_bit && sm==D3D_SHADER_MODEL_6_0) {
+        sprintf_s(compilerOptions, sizeof(compilerOptions),
+                                "-D FALLBACK=%d, -D NON_UNIFORM=%d", 1, 1);
+      }
+      else if (non_uniform_bit || sm==D3D_SHADER_MODEL_6_0){
+        if (non_uniform_bit){
+          sprintf_s(compilerOptions, sizeof(compilerOptions),
+                                "-D NON_UNIFORM=%d",1);
+        } 
+        else{
+          sprintf_s(compilerOptions, sizeof(compilerOptions),
+                                "-D FALLBACK=%d",1);
+        }
+      }
 
-    // Test Compute shader
-    {
-      pShaderOp->CS = pShaderOp->GetString("CS66");
-      std::shared_ptr<ShaderOpTestResult> test = RunShaderOpTestAfterParse(
-          pDevice, m_support, "DynamicResourcesUniformIndexing", nullptr,
-          ShaderOpSet);
+      // by default a root value is added.
+      // remove the root value if this is the non-fallback path
+      if (sm==D3D_SHADER_MODEL_6_6 && non_uniform_bit)
+      {
+        pShaderOp->RootValues.clear();
+      }
 
-      MappedData resultData;
-      test->Test->GetReadBackData("g_result", &resultData);
-      const float *resultFloats = (float *)resultData.data();
+      // Update shader target in xml.
+      for (st::ShaderOpShader &S : pShaderOp->Shaders){
+        if (sm==D3D_SHADER_MODEL_6_0){
+          S.Arguments = compilerOptions;
```

Comment:
You can't just safely assign compilerOptions to S.Arguments here like this.  Make sure you do this instead:
```
  S.Arguments = pShaderOp->GetString(compilerOptions);
```
This will assign a string that pShaderOp owns rather than having it refer to a string on the stack here.